src/pods/import-collection/import-panel.pod.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import classes from './import-panel.pod.module.css';
+import * as canvasVm from '@/core/providers/canvas-schema';
+import * as editTableVm from '../edit-table/edit-table.vm';
+import { doMapOrCreateTable } from '../edit-table/edit-table.business';
+import { mapEditTableVmToTableVm } from '../edit-table/edit-table.mapper';
+import { parseJsonToFieldVm } from './edit-table.business';
+
+interface ImportPanelProps {
+  table?: canvasVm.TableVm;
+  relations: canvasVm.RelationVm[];
+  onSave: (table: canvasVm.TableVm) => void;
+  onClose: () => void;
+}
+
+const defaultJson = JSON.stringify(
+  {
+    _id: { $oid: '67bdea3c01572368ed0b2d81' },
+    room: 'ada-36567',
+    content: '\nHello\n\nworld\n',
+    expireAt: {
+      $date: { $numberLong: '1740499516555' },
+      isNN: true,
+    },
+    __v: { $numberInt: '0' },
+  },
+  null,
+  2
+);
+
+export const ImportPanel: React.FC<ImportPanelProps> = props => {
+  const { onSave, onClose, relations, table } = props;
+  const [jsonContent, setJsonContent] = React.useState<string>(defaultJson);
+  const [jsonError, setJsonError] = React.useState<string | null>(null);
+
+  const [editTable, setEditTable] = React.useState<editTableVm.TableVm>(() =>
+    doMapOrCreateTable(relations, table)
+  );
+
+  const handleSubmit = () => {
+    try {
+      const parsedJson = JSON.parse(jsonContent);
+      const parsedFields = parseJsonToFieldVm(parsedJson);
+      const newTable: editTableVm.TableVm = {
+        ...editTable,
+        fields: parsedFields,
+      };
+      onSave(mapEditTableVmToTableVm(newTable));
+    } catch (error) {
+      setJsonError('El JSON no es válido');
+    }
+  };
+
+  const handleJsonContentChange = (
+    e: React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    const newValue = e.currentTarget.value;
+    setJsonContent(newValue);
+
+    try {
+      JSON.parse(newValue);
+      setJsonError(null);
```

Comment:
One additional case, the JSON maybe valid, but if the user passes an array (e.g. an array with two documents), we should as well treat this as an Error and display a message "Current version only accepts a single document object"

Reply:
Let's move this code to business so we can add unit tests